### PR TITLE
Add (&) to both 7.8.4 and 7.10.2

### PR DESCRIPTION
**Pull request by jystic**
/cc @markhibberd @charleso @tmcgilchrist

### CODE REVIEW

**Comment by jystic** [comment]
Implemention from http://hackage.haskell.org/package/base-4.8.1.0/docs/src/Data.Function.html#%26

File: src/P/Function.hs [hunk 1]
```diff
@@ -15,6 +22,20 @@ import           Data.List (foldr, replicate)
...
 
+#if (__GLASGOW_HASKELL__ >= 710)
+
+import           Data.Function ((&))
+
+#else
+
+-- | '&' is a reverse application operator.  This provides notational
+--   convenience.  Its precedence is one higher than that of the forward
+--   application operator '$', which allows '&' to be nested in '$'.
```
Comment (charleso) [inline src/P/Function.hs]: Are you missing the actual `infix` part?

**Comment by charleso** [comment]
:+1:

File: src/P/Function.hs (hunk 1 above)
Comment (jystic) [inline src/P/Function.hs]: Yes! thanks

Comment (jystic) [inline src/P/Function.hs]: Wow looks like they define a precedence for infix `on`, didn't know that was possible

**Pull request merged**